### PR TITLE
Listeners perf

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This PR introduces a new benchmark focused on listeners creation. It also contains small refactoring and performance improvement. Go over this PR commit-by-commit for easier review

### CODE REVIEW

File: packages/core/test/render3/perf/listeners/index.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+// initialize global state
+resetComponentState();
+
+// create view once so we don't profile first template pass
+createAndRenderLView(null, embeddedTView, viewTNode);
+
+// profile create views (run templates in creation mode)
+console.profile('create listeners');
+for (let i = 0; i < 500000; i++) {
```
Comment (mhevery) [inline packages/core/test/render3/perf/listeners/index.ts]: Hey can we start using [A better way to write perf tests](http://www.typescriptlang.org/play/#code/MYewdgzgLgBAygSQFoFEYF4YEYAMBuAKANElgEMAnCsgTwHEKQB3KACwxmAoFMypuAQtzDBWAW0oBrABQByStRowA5oxatZAGmw5du7QFYcASkIEANt1gBLMABNuADwAq1sdw4LaDZmzkBtWwdHAF1ZUwImVmtLGGkgp1d3ADoKAFcwaWNjGABvAhgYKJjuePtEt1LsvILCmEtyKloOfxDCOpgAMxAKaQaYSXR8AZgAHnhkFDwBgGoZnPyOjq8af0kQjnxawoBfWr2ds36ABzSIViSPTBWfdTlT8-DCYstpB4vK1IyshdqX0velx+NQ6-RWLTa2y6PT6VgGQ2mkjGE1QiLmvyWdRWyXe0hM7Tqe12BEORBuajYqW4xx6UGkZEsFCgESIAHp2RzOVzuRyYAIUAA5ADCAAkALIAQQASgBpGBCgDyABE0Dy1eqiLZ+BROmRgB4hCJxFIQXEAG7cCgQazgAVkdwALhg0AotmUxidAAVGJ0SgSeDSmdJOmAAPxO6QB2lOl1unLoAB8MDNIGsdg9ydTdkIey1lt1+pg3pAvtiiziGYARiAQJYyGACccfSU7Y7nVBXWBlP7vlWa3WG7VK9xoJcnWA0mJhxQc0ROhlgFAbWBODw+IJhKIJBQZMOjdvJK3uDGO27tNZtXxl0KQBkoOPJ9PtOlIA6J1PLRwjBnDVuTaaSGgGAmxLEoIC9ZtLFaCEzEKQDYD3P8dw4ecRCXcA+U3Y0dzeSDuCPE9O3dCDQLLKF+gvS0r3AG870tB8PwoDhKOodCwFosAoAJODwCAkDS2PIs8JQhc2KE0iqlNDprE6OIWOo9jb04z90FUmATCkpZ5LYjjtWYy8dKUrioQ6HgoDSCgV11cwIG4bjCRgbgbI8ctMW068jMtABaLz7NMqwLJXDs0jskyYCJByyAgcSBPs-pzCiqBLmge1jk2OK4RfXSOBfCB7P4kovhXTBUMXZcsk0njSBgMBmA4Y58x6CQRG4ZJaqYLI-JgGS4gS0dKhSsQ0oAQjUjTXKWeCYCXdwOHamAvPqRLkqgVKusKArLGSYd+tmzAxT4VhkjEWxcIk7aRySyptBm7gIkxCLQWWgbVqGubmC6nrIwybLVMwcawsKLKjJyjI8sBmAzMCroGVsrqdkc5zKrqYG7x89bIYCyzpvSUKHqhR7gLwnE8KPerSftPG6k21qdquvaYAFR9LWSSUAA0AH04AlAAxFAOYQAVnBQOgUClfK8IgHEzlYM6BPulGsZXGmCQRqLMP3KRYJgRDsMkKlA1gErRPKkNwziKMmUIuMMCTFM0wxOpbPp7hbzpCrE2RqbdWgS7yYkqWeDsNJ9WkXDuDNG0zhIgTtGACyeE4mOSnjJMJv88zsfDyPbylunLmReOqGEKALt2u6YFDImI6j6KnSLxPjIehWOmDFcAB9284Xja1a8wQHdaQAANfz1p0ABJcl1g8jx2AAdMBJ5pqWJGOMOaeTyxU+RqqgNujg5cK-PKhgVluoMjy7xgAAqHRdGMZIoBAHnrEcbg7GkABmFvJt42AGooPqTiB9cA4AWkTc6x9Zpnx9vwaAZcXY3zvg-J+L834f3xBDKG2Mh6FCXsTGms8J65FugjM4cQl6WiAVAHYABSYwQ94YPwAFapkyLIBe4QdhDx-jsBWEVsErmnlrEkBAgA)

Comment (pkozlowski-opensource) [reply packages/core/test/render3/perf/listeners/index.ts]: Done, it is using the micro-harness right now.

**Review by mhevery — APPROVED** [review]

File: packages/core/test/render3/perf/listeners/index.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import {ɵɵelementEnd, ɵɵelementStart} from '../../../../src/render3/instructions/element';
+import {ɵɵlistener} from '../../../../src/render3/instructions/listener';
+import {createTNode, createTView} from '../../../../src/render3/instructions/shared';
+import {RenderFlags} from '../../../../src/render3/interfaces/definition';
+import {TNodeType, TViewNode} from '../../../../src/render3/interfaces/node';
+import {resetComponentState} from '../../../../src/render3/state';
+import {createBenchmark} from '../micro_bench';
+import {createAndRenderLView} from '../setup';
+
+const _c0 = [3, 'click', 'input'];
```
Comment (kara) [inline packages/core/test/render3/perf/listeners/index.ts]: Nit: could we add a block comment with what the template would look like?

**Review by kara — APPROVED** [review]
LGTM, one nit

File: packages/core/test/render3/perf/listeners/index.ts (hunk 2 above)
Comment (pkozlowski-opensource) [reply packages/core/test/render3/perf/listeners/index.ts]: Totally! Was adding it in all other benchmarks, not sure why I've skipped it here...

**Comment by matsko** [comment]
Presubmit = http://test/OCL:268267458:BASE:268267476:1568139502258:e26a5e3c

Ivy Presubmit = http://test/OCL:268267458:BASE:268267562:1568139524254:4de75b96

**Comment by matsko** [comment]
Presubmit = https://test.corp.google.com/ui#id=OCL:268267458:BASE:268477361:1568219981221:2dd2487a

Ivy Presubmit = https://test.corp.google.com/ui#id=OCL:268267458:BASE:268477897:1568220142527:2548025c

**Pull request closed without merging**